e2e_tests/card_test.py

```diff
@@ -0,0 +1,117 @@
+import os
+import unittest
+from datetime import date, timedelta
+from api.unit import Unit
+from models.card import CreateIndividualDebitCard, PatchIndividualDebitCard
+from models.account import *
+from models.application import CreateIndividualApplicationRequest
+
+
+class CardE2eTests(unittest.TestCase):
+    token = os.environ.get("token")
+    client = Unit("https://api.s.unit.sh", token)
+    card_types = ["individualDebitCard", "businessDebitCard", "individualVirtualDebitCard", "businessVirtualDebitCard"]
+
+    def get_card_id(self, opt_dict: dict[str, str]):
+        response = self.client.cards.list()
+        for card in response.data:
+            cond = True
+            for key, value in opt_dict.items():
+                if getattr(card, key) != value:
+                    cond = False
+
+            if cond:
+                return card.id
+
+        return ""
```

Comment:
this is not clear!

1. function name - get a card id out of what? 
2. parameters - opt_dict is not a meaningful name and I still don't know what thats means.

while you actually want to find the id of a random card answering to your criteria!

```suggestion
    def find_card_id(self, criteria: dict[str, str]):
        response = self.client.cards.list()
        filtered = list(filter(lambda c: ...))
        return filtered[0] if len(...) > 0 else ""
```